xdsl/utils/lexer.py

```diff
@@ -107,9 +108,19 @@ def len(self):
     def text(self):
         return self.input.content[self.start : self.end]
 
+    def get_line_start_and_index(self) -> LineStartAndIndex:
+        """
+        The start offset and 1-index of position in a file.
+        The line index is offset by `line_offset`, which is non-0 when the input file is
+        split.
+        """
+        line_start = self.input.get_start_of_line(self.start)
+        line_index_in_source = self.input.content.count("\n", 0, line_start) + 1
+        line_index = line_index_in_source + self.line_offset
+        return LineStartAndIndex(line_start, line_index)
+
     def get_line_col(self) -> tuple[int, int]:
-        info = self.input.get_lines_containing(self)
-        _lines, offset_of_first_line, line_no = info
+        offset_of_first_line, line_no = self.get_line_start_and_index()
```

Comment:
Can named tuples be destructured like this?

Reply:
Yep, this is an argument against using them for some people, but I find it convenient.